packages/store-ui/src/hooks/useSlider/useSlider.ts

```diff
@@ -0,0 +1,226 @@
+import type { Dispatch } from 'react'
+import { useReducer } from 'react'
+import type { SwipeableProps } from 'react-swipeable'
+import { useSwipeable } from 'react-swipeable'
+
+export type SlideDirection = 'next' | 'previous'
+
+interface NextPageAction {
+  type: 'NEXT_PAGE'
+}
+
+interface PreviousPageAction {
+  type: 'PREVIOUS_PAGE'
+}
+
+interface GoToPageAction {
+  type: 'GO_TO_PAGE'
+  payload: {
+    pageIndex: number
+    shouldSlide: boolean
+  }
+}
+
+interface StopSlideAction {
+  type: 'STOP_SLIDE'
+}
+
+export type Action =
+  | NextPageAction
+  | PreviousPageAction
+  | StopSlideAction
+  | GoToPageAction
+
+export type SliderDispatch = Dispatch<Action>
+
+export interface SliderState {
+  /**
+   * The `currentItem` in a Slider with multiple items in a single page is
+   * always **the one with the lowest index** in the current page.
+   */
+  currentItem: number
+  /** Currently active page */
+  currentPage: number
+  /**
+   * Whether or not the Slider is currently sliding. This is useful to
+   * manipulate the `transition` property in a component.
+   */
+  sliding: boolean
+  /** The direction in which the Slider is sliding. */
+  slideDirection: SlideDirection
+  /** The total number of unique items in the slider. */
+  totalItems: number
+  /** The number of items in a single page. */
+  itemsPerPage: number
+  /** The total number of pages in the slider. */
+  totalPages: number
+  /** Whether or not the slider is infinite. */
+  infinite: boolean
+}
+
+export interface UseSliderArgs extends SwipeableProps {
```

Comment:
What do you think about putting this stick together with the useSlider function?

Reply:
Do you mean declaring it closer to the `useSlider` function? Sure! 

I left it up here just to have all interfaces declared first, I guess it's just something I'm used to doing 😅  I'll move it closer to the function!